### DISCUSSION TITLE
MoviePilot的历史记录希望可以增加一键删除功能

### DISCUSSION THREAD

**Discussion by 17612587** [jxxghp/MoviePilot#1499] [建议]
MoviePilot的历史记录希望可以增加一键删除功能，每次需要重新刮削转移的时候手动删除一页100条的删除，选择ALL直接卡死，现在短剧比较火爆，一般短剧动不动100集。我历史记录6万多条，删一整天都删除不玩，希望可以出一个一件删除功能

**Comment by it985** [comment]
+1，我这边80w+的短剧记录，all就oom，只有1k最高可选，删到手酸都删不完

**Comment by pushduck** [comment]
> MoviePilot的历史记录希望可以增加一键删除功能，每次需要重新刮削转移的时候手动删除一页100条的删除，选择ALL直接卡死，现在短剧比较火爆，一般短剧动不动100集。我历史记录6万多条，删一整天都删除不玩，希望可以出一个一件删除功能

登录mp 浏览器控制台执行
```javascript
// 起始ID
const maxId = 4321;
const token = `Bearer xxx`;
const host = "//127.0.0.1:8080"

async function deleteHistory() {
  for (let id = maxId; id >= 0; id--) {
    try {
      const response = await fetch(`${host}/api/v1/history/transfer?deletesrc=false&deletedest=false`, { 
        method: 'DELETE',
        headers: {
          'accept': 'application/json, text/plain, */*',
          'accept-language': 'zh-CN,zh;q=0.9',
          'authorization': token,
          'cache-control': 'no-cache',
          'content-type': 'application/json',
          'pragma': 'no-cache'
        },
        referrerPolicy: 'no-referrer',
        body: JSON.stringify({ id: id }),
        mode: 'cors',
        credentials: 'include'
      });

      const result = await response.json();
      console.log(`Deleted item with id ${id}:`, result);
    } catch (error) {
      console.error(`Failed to delete item with id ${id}:`, error);
    }
  }
}

deleteHistory();
```

**Reply by 38348181** [reply]
请问这个如何使用？我直接复制运行显示错误，有哪几个地方需要修改吗

**Reply by wikrin** [reply]
> > MoviePilot的历史记录希望可以增加一键删除功能，每次需要重新刮削转移的时候手动删除一页100条的删除，选择ALL直接卡死，现在短剧比较火爆，一般短剧动不动100集。我历史记录6万多条，删一整天都删除不玩，希望可以出一个一件删除功能

[API文档](https://api.movie-pilot.org/#/history/empty_transfer_history_api_v1_history_empty_transfer_get)在mp保持登录的页面控制台输入
```
MoviepilotAPI.get('history/empty/transfer')
```
执行就清空了

**Comment by 38348181** [comment]
遇到一样的问题了，请问现在有办法解决吗

**Comment by TelBotDev** [comment]
同样的疑惑，之前用SQLite的时候有一个“历史记录清理”插件，现在升级到了PostgreSQL，用什么插件清理记录了呢

**Comment by gitcook** [comment]
curl -X GET \
  'http://192.168.8.110:3000/api/v1/history/empty/transfer' \
  -H 'Authorization: Bearer eyJhbGciOiJIUxxxxxxxxxxxxx'
随便点一个接口请求,在 Request Headers 中找到：Authorization: Bearer eyJhbGciOiJ，复制粘贴
在linux终端运行就全删了

**Reply by cddjr** [reply]
现在支持通过apikey操作这些接口了，会更简单
